terraform/source.go

```diff
@@ -195,7 +199,57 @@ func ToSourceUrl(source string, workingDir string) (*url.URL, error) {
 		return nil, errors.WithStackTrace(err)
 	}
 
-	return parseSourceUrl(rawSourceUrlWithGetter)
+	sourceUrl, err := parseSourceUrl(rawSourceUrlWithGetter)
```

Comment:
Sorry I didn't notice this before, but I'm noticing you're passing in `rawSourceUrlWithGetter` to this method. The `WithGetter` part suggests to me this is a "getter" as parsed by the `go-getter` library. is it possible that the getter is _already_ set to the value we need, and we just need to prepend it at the end? In other words, we don't need to write our own custom logic to figure out the getter?

Reply:
Understand your thought, because I was thinking the same, since this is the same Golang package, it would be logical that the result from `getter.Detect` could be immediately used for `getter.Get`, but unfortunately this is not the case. I tested, `getter.Detect` does prepend a source type if we only pass URL without scheme:

 `github.com/gruntwork-io/terraform-aws-lambda` -> `git::https://github.com/gruntwork-io/terraform-aws-lambda.git`

and doesn't if there's the scheme:

`https://github.com/gruntwork-io/terraform-aws-lambda` -> `https://github.com/gruntwork-io/terraform-aws-lambda.git`

I also tested to pass `https://github.com/gruntwork-io/terraform-aws-lambda` to `getter.Get`, and got the error

```
'https://github.com/gruntwork-io/terraform-aws-lambda': bad response code: 404
```

Reply:
The only way is if we remove the schema before passing it to `Detect`.

Reply:
There is a new version 2.2.1 of `go-getter`, but it is very different from the 1.7.1 version we are using now.
We'll have to rewrite the code to use it. I don’t yet know how it behaves with the http scheme. Should I check this?

Reply:
Thanks for looking into it. Prob not worth a big upgrade now. Let's go with what you have here.